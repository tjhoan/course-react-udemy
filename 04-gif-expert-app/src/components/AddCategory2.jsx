import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory2 = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const inputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 2) return;
    onNewCategory(inputValue.trim());
    setInputValue('');
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input type="text" placeholder="Buscar Gifs" value={inputValue} onChange={inputChange} />
    </form>
  );
};

AddCategory2.propTypes = {
  onNewCategory: PropTypes.func.isRequired
};
