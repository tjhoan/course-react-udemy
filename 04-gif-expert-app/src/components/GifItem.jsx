import PropTypes from 'prop-types';

export const GifItem = ({ id, title, url }) => {
  return (
    <div className="card" key={id}>
      <img className="card img" src={url} alt={title} />
      <p className="card p">{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};
