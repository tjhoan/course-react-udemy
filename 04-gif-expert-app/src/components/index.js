export * from './AddCategory2';
export * from './GifGrid';
export * from './GifItem';
