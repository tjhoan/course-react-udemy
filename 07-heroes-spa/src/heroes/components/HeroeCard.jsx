import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export const HeroeCard = ({
  id,
  superhero,
  publisher,
  alter_ego,
  first_appearance,
  characters
}) => {
  const heroImageUrl = `/assets/heroes/${id}.jpg`;

  return (
    <div className="col animate__animated animate__fadeIn">
      <div className="card h-100" style={{ maxWidth: '200px' }}>
        <img
          src={heroImageUrl}
          alt={superhero}
          className="card-img-top"
        />
        <div className="card-body bg-dark text-white">
          <h4 className="card-title mb-3">{superhero}</h4>
          <p className="card-text">{alter_ego}</p>
          <p className="card-text">{publisher}</p>
          <p className="card-text">{first_appearance}</p>
          {alter_ego !== characters && <p className="card-text">{characters}</p>}
          <Link to={`/hero/${id}`} className="btn btn-warning">
            MÃ¡s Info...
          </Link>
        </div>
      </div>
    </div>
  );
};

HeroeCard.propTypes = {
  id: PropTypes.string.isRequired,
  superhero: PropTypes.string.isRequired,
  publisher: PropTypes.string.isRequired,
  alter_ego: PropTypes.string.isRequired,
  first_appearance: PropTypes.string.isRequired,
  characters: PropTypes.string.isRequired
};
