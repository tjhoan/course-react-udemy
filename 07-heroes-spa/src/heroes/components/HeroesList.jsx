import { getHeroesByPublisher } from '../helpers';
import PropTypes from 'prop-types';
import { HeroeCard } from './';

export const HeroesList = ({ publisher }) => {
  const heroes = getHeroesByPublisher(publisher);

  return (
    <>
      <div className="row row-cols-1 row-cols-sm-3 row-cols-md-4 row-cols-lg-auto g-3 mt-2 mb-5">
        {heroes.map((hero) => (
          <HeroeCard key={hero.id} {...hero} />
        ))}
      </div>
    </>
  );
};

HeroesList.propTypes = {
  publisher: PropTypes.string.isRequired
};
