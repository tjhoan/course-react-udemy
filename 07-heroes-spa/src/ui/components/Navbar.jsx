import { Link, NavLink, useNavigate } from 'react-router-dom';

export const Navbar = () => {
  const navigate = useNavigate();

  const onLogout = () => {
    navigate('/login', {
      replace: true
    });
  };

  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark p-2 fixed-top">
      <Link className="navbar-brand" to="/">
        Asociaciones
      </Link>

      <div className="navbar-collapse">
        <div className="navbar-nav">
          <NavLink
            to="/marvel"
            className={({ isActive }) => {
              return `nav-item nav-link ${isActive ? 'text-white active' : 'text-secondary'}`;
            }}
          >
            Marvel
          </NavLink>

          <NavLink
            to="/dc"
            className={({ isActive }) => {
              return `nav-item nav-link ${isActive ? 'text-white active' : 'text-secondary'}`;
            }}
          >
            DC
          </NavLink>
        </div>
      </div>

      <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end">
        <ul className="navbar-nav ml-auto">
          <span className="nav-item nav-link text-secondary">Jhoan</span>
          <button className="nav-item nav-link btn" onClick={onLogout}>
            Logout
          </button>
        </ul>
      </div>
    </nav>
  );
};
