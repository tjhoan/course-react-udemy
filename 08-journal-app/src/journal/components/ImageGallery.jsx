import { ImageList, ImageListItem, useMediaQuery } from '@mui/material';
import { itemData } from '../../data/itemData';

export const ImageGallery = () => {
  const isExtraSmallScreen = useMediaQuery('(max-width:700px)');
  const isSmallScreen = useMediaQuery('(max-width:850px)');
  const isMediumScreen = useMediaQuery('(max-width:100px)');
  const isLargeScreen = useMediaQuery('(max-width:1300px)');

  const cols = isExtraSmallScreen
    ? 2
    : isSmallScreen
    ? 3
    : isMediumScreen
    ? 4
    : isLargeScreen
    ? 5
    : 7;

  return (
    <ImageList sx={{ width: '100%', height: '100%' }} variant="masonry" cols={cols} gap={4}>
      {itemData.map((item) => (
        <ImageListItem key={item.img}>
          <img
            src={`${item.img}?w=164&h=164&fit=crop&auto=format`}
            alt={item.title}
            loading="lazy"
            style={{
              borderRadius: 5,
              width: '100%',
              height: 'auto',
              objectFit: 'cover'
            }}
          />
        </ImageListItem>
      ))}
    </ImageList>
  );
};
