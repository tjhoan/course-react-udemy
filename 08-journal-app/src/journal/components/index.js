export * from './ImageGallery';
export * from './SideBarItem';
export * from './SideBar';
export * from './Navbar';
