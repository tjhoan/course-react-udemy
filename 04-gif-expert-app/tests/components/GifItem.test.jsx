import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

const props = {
  title: 'Hola, soy Goku',
  id: 'Gt23ioj23gvhgv43',
  url: 'https://qweqweqwe.png'
};

describe('Pruebas en <GifItem />', () => {
  test('Debe de hacer match con el snapshot', () => {
    const { container } = render(<GifItem {...props} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe de coincidir el tÃ­tulo del props y el del componente', () => {
    const { getByText } = render(<GifItem {...props} />);
    expect(getByText(props.title)).toBeTruthy();
  });

  test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem {...props} />);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(`${props.url}/`);
    expect(alt).toBe(props.title);
  });

  test('Debe de mostrar el titulo del componente', () => {
    render(<GifItem {...props} />);
    expect(screen.getByText(props.title)).toBeTruthy();
  });
});
