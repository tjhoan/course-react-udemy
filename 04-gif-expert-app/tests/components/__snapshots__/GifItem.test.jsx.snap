// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola, soy Goku"
      class="card img"
      src="https://qweqweqwe.png"
    />
    <p
      class="card p"
    >
      Hola, soy Goku
    </p>
  </div>
</div>
`;
